odoo/addons/base/tests/test_expression.py

```diff
@@ -795,6 +795,39 @@ def test_filtered_domain_order(self):
         countries = countries.browse(reversed(countries._ids))
         self.assertEqual(countries.filtered_domain(domain)._ids, countries._ids)
 
+    def test_filter_domain_with_archived_records(self):
+        """ Test that filtering a domain on archived records works as expected """
+        partner_obj = self.env["res.partner"]
+        partner = partner_obj.create({"name": "Foo"})
+
+        init_domain = [("id", "in", partner.ids), ("active", "in", (True, False))]
+        domain = [("parent_id", "child_of", partner.id)]
+        self.assertTrue(partner.filtered_domain(domain))
+        filtered_partner = partner.with_context(active_test=False).filtered_domain(domain)
+        self.assertIn(partner, filtered_partner)
+        self.assertIn(partner_obj.search(init_domain + domain), filtered_partner)
```

Comment:
This test has no value, since you don't check whether `partner_obj.search(init_domain + domain)` and `filtered_partner` are equal, which is the whole point of the discussion.

For a more meaninful test, add children to partner (active and inactive)
```python
children = partner_obj.create([
    {'name': 'Bar', 'parent_id': partner_id, 'active': True},
    {'name': 'Baz', 'parent_id': partner_id, 'active': False},
])
records = partner + children
```
and then compare the results for a domain like `[('child_ids.name', '=', 'Baz')]`.